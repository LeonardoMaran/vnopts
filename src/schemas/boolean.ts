import { Schema, SchemaParameters } from '../schema';
import { ValidateResult } from '../types';

export class BooleanSchema extends Schema<boolean, SchemaParameters<boolean>> {
  public expected() {
    return 'true or false';
  }
  public validate(value: unknown): ValidateResult {
    return typeof value === 'boolean';
  }
}
