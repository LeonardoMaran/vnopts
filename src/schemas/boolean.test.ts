import { eachHandler } from '../../tests/__helpers__/utils';
import { BooleanSchema } from './boolean';

const name = '<key>';
const value = true;

describe.each`
  parameters | input                    | output               | hasWarnings
  ${{}}      | ${{ [name]: value }}     | ${{ [name]: value }} | ${false}
  ${{}}      | ${{ [name]: 'invalid' }} | ${Error}             | ${false}
`('', eachHandler<BooleanSchema>(BooleanSchema, { name }));
