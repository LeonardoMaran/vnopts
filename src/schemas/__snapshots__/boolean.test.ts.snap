// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema {"name":"<key>"} with input {"<key>":"invalid"} 1`] = `"Invalid \\"<key>\\" value. Expected true or false, but received \\"invalid\\"."`;
